tests: cmake: actually test SBOM files generation

kartben opened the pull request:
The current test was just a basic sanity check.
This adds actual validation of the SPDX files being generated as being spec-compliant.

Will start adding more tests over time to actually validate _contents_ of the file is OK too, not just the form :)

kartben committed:
tests: cmake: actually test SBOM files generation

The current test was just a basic sanity check.
This adds actual validation of the SPDX files being generated as
being spec-compliant.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

kartben commented:
Need to figure out why CI is seemingly not picking it up, but will happily take feedback in the meantime.
BTW it looks like the `twister-build` step doesn't `pip install -r scripts/requirements-actions.txt` - shouldn't it? @nashif

kartben committed:
ci: twister: install Python dependencies in twister-build step

This ensures we catch up on latest requirements in case Docker image is
lagging behind.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

pdgendt commented on .github/workflows/twister.yaml:
```diff
@@ -210,6 +210,10 @@ jobs:
           echo "github.base_ref: ${{ github.base_ref }}"
           echo "github.ref_name: ${{ github.ref_name }}"
 
+      - name: Install Python packages
+        run: |
+          pip install -r scripts/requirements-actions.txt --require-hashes
```
Need to setup Python too?

```suggestion
    - name: Set up Python
      uses: actions/setup-python@a26af69be951a213d495a4c3e4e4022e16d87065 # v5.6.0
      with:
        python-version: 3.12
        cache: pip
        cache-dependency-path: scripts/requirements-actions.txt

    - name: Install Python packages
      run: |
        pip install -r scripts/requirements-actions.txt --require-hashes

```

kartben replied:
I dunno, I guess -- I never understand how this setup step is supposed to work (esp. since here we already have python in the container)

stephanosio replied:
> Need to setup Python too?

No, we should be using the Python included in the CI Docker image. If we need to change the Python version, the Docker image should be updated instead.

kartben committed:
doc: develop: west: document spdx not being supported for native_sim

Generating SPDX SBOMs for the ``native_sim`` platform has propably
never worked, but at least now it will be documented.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

kartben commented:
So it turns out that SBOM generation for `native_sim` is kind of non-functional, probably always has been. I don't think it should prevent this from going in though, so I've opened #91785 and added to this PR a brief note in the documentation describing the limitation.

CI now happy with all SPDX files for 2.2 and 2.3 for hello_world on qemu_cortex_m3 being flagged as valid.
